pipeline.libsonnet

```diff
@@ -0,0 +1,187 @@
+local windows_pipe = '\\\\\\\\.\\\\pipe\\\\docker_engine';
+local windows_pipe_volume = 'docker_pipe';
+local test_pipeline_name = 'testing';
+
+local windows(os) = os == 'windows';
+
+local golang_image(os, version) =
+  'golang:' + '1.11' + if windows(os) then '-windowsservercore-' + version else '';
+
+{
+  test(os='linux', arch='amd64', version='')::
+    local golang = golang_image(os, version);
+    {
+      kind: 'pipeline',
+      name: test_pipeline_name,
+      platform: {
+        os: os,
+        arch: arch,
+        version: if std.length(version) > 0 then version,
+      },
+      steps: [
+        {
+          name: 'vet',
+          image: golang,
+          pull: 'always',
+          environment: {
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go vet ./...',
+          ],
+        },
+        {
+          name: 'test',
+          image: golang,
+          pull: 'always',
+          environment: {
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go test -cover ./...',
+          ],
+        },
+      ],
+    },
+
+  build(name, os='linux', arch='amd64', version='')::
+    local is_windows = windows(os);
+    local tag = if is_windows then os + '-' + version else os + '-' + arch;
+    local file_suffix = std.strReplace(tag, '-', '.');
+    local volumes = if is_windows then [{ name: windows_pipe_volume, path: windows_pipe }] else [];
+    local golang = golang_image(os, version);
+    local plugin_repo = 'plugins/' + std.split(name, '-')[1];
+    local extension = if is_windows then '.exe' else '';
+    {
+      kind: 'pipeline',
+      name: tag,
+      platform: {
+        os: os,
+        arch: arch,
+        version: if std.length(version) > 0 then version,
+      },
+      steps: [
+        {
+          name: 'build-push',
+          image: golang,
+          pull: 'always',
+          environment: {
+            CGO_ENABLED: '0',
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go build -v -ldflags "-X main.build=${DRONE_BUILD_NUMBER}" -a -o release/' + os + '/' + arch + '/' + name + extension,
+          ],
+          when: {
+            event: ['push', 'pull_request'],
+          },
+        },
+        {
+          name: 'build-tag',
+          image: golang,
+          pull: 'always',
+          environment: {
+            CGO_ENABLED: '0',
+            GO111MODULE: 'on',
+          },
+          commands: [
+            'go build -v -ldflags "-X main.version=${DRONE_TAG##v} -X main.build=${DRONE_BUILD_NUMBER}" -a -o release/' + os + '/' + arch + '/' + name + extension,
+          ],
+          when: {
+            event: ['tag'],
+          },
+        },
+        {
+          name: 'executable',
+          image: golang,
+          pull: 'always',
+          commands: [
+            './release/' + os + '/' + arch + '/' + name + extension + ' --help',
```

Comment:
Does this also work on Windows? Or do we need to rewrite it for Windows?

Reply:
Yea it works